packages/catalog-process/src/services/readModelService.ts

```diff
@@ -33,16 +34,42 @@ function arrayToFilter<T, F extends object>(
   return array.length > 0 ? f(array) : undefined;
 }
 
+async function getTotalCount(
+  query: AggregationCursor<Document>
+): Promise<number> {
+  const data = await query.toArray();
+  const result = z.array(z.object({ count: z.number() })).safeParse(data);
+
+  if (result.success && result.data.length > 0) {
+    return result.data[0].count;
+  }
+
+  logger.warn(
+    `Unable to get total count from aggregation pipeline: result ${JSON.stringify(
+      result
+    )} - data ${JSON.stringify(data)} `
+  );
+  return 0;
```

Comment:
for the count, I agree with logging a warning and discarding the error: it seems to match the Scala code

for the main results, I think the Scala code has the `Future` fail: so it's good to log an error but maybe we should throw it too, rather than returning an empty result (it's not super clear from the Scala code, so maybe we can open a card to discuss it later)